sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -279,34 +280,69 @@ case class AdaptiveSparkPlanExec(
       prefix: String = "",
       addSuffix: Boolean = false,
       maxFields: Int,
-      printNodeId: Boolean): Unit = {
-    super.generateTreeString(depth,
+      printNodeId: Boolean,
+      indent: Int = 0): Unit = {
+    super.generateTreeString(
+      depth,
       lastChildren,
       append,
       verbose,
       prefix,
       addSuffix,
       maxFields,
+      printNodeId,
+      indent)
+    generateTreeStringWithHeader(
+      if (isFinalPlan) "Final Plan" else "Current Plan",
```

Comment:
> Why it's Current Plan not Final Plan in EXPLAIN FORMATTED

@cloud-fan I think it is from here.
@allisonwang-db Could you update the PR description to explain the difference?

Reply:
Since `EXPLAIN FORMATTED` will not actually execute the query, we use `Current Plan` to indicate the *current* query plan, and when the current adaptive query plan is final, we will use `Final Plan`.